components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -850,6 +861,49 @@ public static boolean isValidWSAPI(APIDTO api) {
         return isValid;
     }
 
+    /**
+     * Validate sandbox and production endpoint URLs.
+     *
+     * @param apiDto API DTO of the API
+     * @return validity of URLs found within the endpoint configurations of the DTO
+     */
+    public static boolean validateEndpoints(APIDTO apiDto) {
```

Comment:
Shall we move this logic into APIM core level? Usually it is not good to perform business logic in REST API components.
If we move this to core level, we can reuse this in other cases too in the future.